src/trie.rs

```diff
@@ -160,16 +161,22 @@ where
                         let value_option = branch.borrow().value.clone();
                         if let Some(value) = value_option {
                             return Some((self.nibble.encode_raw().0, value));
-                        }
-                        else {
+                        } else {
                             continue;
                         }
                     }
 
                     (TraceStatus::Doing, Node::Hash(ref hash_node)) => {
-                        if let Ok(n) = self.trie.recover_from_db(&hash_node.borrow().hash.clone()) {
+                        let node_hash = hash_node.borrow().hash.clone();
+                        if let Ok(n) = self.trie.recover_from_db(&node_hash) {
                             self.nodes.pop();
-                            self.nodes.push(n.into());
+                            match n {
+                                Some(node) => self.nodes.push(node.into()),
+                                None => {
+                                    warn!("Trie node with hash {:?} is missing from the database. Skipping...", &node_hash);
```

Comment:
It's ok to just log a warning here, rather than returning an error?

Reply:
Hm, I guess it depends. It might be valuable to trace through the existing database as much as possible, minus any missing data. I don't have a strong sense of when this tracing is used. I'll think about it some more.